### ISSUE TITLE
Audio not playing on Android release bundles

### ISSUE THREAD

**Issue by tristan-nel** [react-native-kit/react-native-track-player#811]
# Configuration

```bash
$ npx react-native info
System:
    OS: Linux 4.15 Ubuntu 18.04.3 LTS (Bionic Beaver)
    CPU: (4) x64 Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
    Memory: 3.06 GB / 15.60 GB
    Shell: 4.4.20 - /bin/bash
  Binaries:
    Node: 12.11.1 - ~/.nvm/versions/node/v12.11.1/bin/node
    npm: 6.11.3 - ~/.nvm/versions/node/v12.11.1/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    Android SDK:
      API Levels: 28
      Build Tools: 28.0.3, 29.0.2
      System Images: android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 At
om_64
  npmPackages:
    @react-native-community/cli: ^2.9.0 => 2.9.0 
    react: 16.8.6 => 16.8.6 
    react-native: ^0.60.5 => 0.60.5 
  npmGlobalPackages:
    react-native-rename: 2.4.1
```

```bash
$ npm react-native-track-player -v
6.11.3
```

# Issue

Attempting to play a .ogg audio stream from a remote source (either HTTP or HTTPS) is failing on the release build variants (APK/bundle) on Android (both devices and emulator). Audio plays perfectly fine on debug builds (both devices and emulator).

# Code

```javascript
  async function initPlayer() {
    if (debug) console.log('init');
    await TrackPlayer.setupPlayer({});
    TrackPlayer.updateOptions({
      capabilities: [TrackPlayer.CAPABILITY_PLAY, TrackPlayer.CAPABILITY_PAUSE],
    });
    if (!playerInit) setPlayerInit(true);
  }

  function loadTracks() {
    if (debug) console.log('loadTracks');
    // TODO: May need adjustments
    TrackPlayer.reset();
    try {
      console.warn(channelData.url);
      TrackPlayer.add({
        id: '1',
        url: channelData.url, // [Liquidsoap -> Icecast stream]  http(s)://IP_ADDR:PORT/radio.ogg
        contentType: 'audio/ogg',
        title: 'MiRadio Stream',
        artist: 'MiRadio',
      });
    } catch (e) {
      console.error(e);
    }

    if (!tracksLoaded) setTracksLoaded(true);
  }

  function playPause() {
    if (debug) console.log('playPause');
    if (playerInit && dataLoaded) {
      setIsPlaying(!isPlaying);
      if (isPlaying) {
        TrackPlayer.pause();
      } else {
        TrackPlayer.play();
      }
    }
  }

  function livePlay() {
    if (!isPlaying) {
      playPause();
      loadTracks();
      playPause();
    }
  }
```

**Comment by HectorFerris** [comment]
Same problem!

**Comment by tristan-nel** [comment]
That solution isn't working for me - and as far as I can tell, it should only be an issue for HTTP - mine is failing to produce audio in release builds for both HTTP and HTTPS

**Comment by tristan-nel** [comment]
@HectorFerris you were on the right track, I think.

New Android OS's do not like HTTP. But adding clearText Permission in AndroidManifest.xml did not work for me.

What _did_ work was I had to serve my audio stream via HTTPS.

First I had to install Icecast2 from Xiph.org directly, so that is builds with SSL support on my Ubuntu 18.04 LTS VM.

Next I had to get a free domain (my.freenom.com), wait about 24 hours for it to propagate (test with `ping EXAMPLE.com` to see when it's active). I also had to add the `A` (server IP) and `CNAME` (MY_DOMAIN) details under "Services / My Domain / Manage domain / Manage Freenom DNS"

```
Name    Type   TTL     Target
        A      3600    IP.ADD.RE.SS
WWW     CNAME  3600    MY_DOMAIN.TK
```

Then I had to use Let'sEncrypt (certbot) to generate a trusted certifcate for my domain on the server which serves the audio stream via Icecast2 (stream produced by Liquidsoap).

Next I had to combine the fullchain cert and private key together (in that order) into a single file:
```bash
cd
mkdir ssl
cd ssl
touch icecast2.pem
sudo cat /etc/letsencrypt/live/MY_DOMAIN/fullchain.pem >> icecast2.pem
sudo cat /etc/letsencrypt/live/miradio-test.tk/privkey.pem >> icecast2.pem
```

Then I had to configure icecast2 to provide **both** an HTTP and an HTTPS listen-socket:

```
sudo nano /etc/icecast2/icecast.xml
```

Then have something like this:
```xml
<listen-socket>
    <port>8000</port>
</listen-socket>
<listen-socket>
    <port>8001</port>
    <ssl>1</ssl>
</listen-socket>
...
<ssl-certificate>~/ssl/icecast2.pem</ssl-certificate>
```

This provides the icecast output streams to both HTTP and HTTPS (which is for some reason required, otherwise liquidsoap streams don't get served by icecast2).

Finally `sudo systemctl restart icecast2`, set my audio stream URL for react-native-track-player to `https://MY_DOMAIN.TK:8001/radio.ogg`, and finally I have audio playing in the release build of my React Native app :)
